### PR TITLE
fix: [PL-29433]: Fix Error response handling for ListFilesInCommit SCM API

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes
Currently, the API throws Runtime exception for all error cases. Ideally, it should return the HTTP response code from git provider and its corresponding error message as proper GRPC response. Fixed it now.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39453)

### CODE REVIEW

**Review by adivishy1 — APPROVED** [review]
> Reviewed 2 of 2 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39453)* status: 13 file(s) to review. 1 of 2 approvals obtained.

File: product/ci/scm/file/file.go [hunk 1]
```diff
@@ -392,7 +392,14 @@ func FindFilesInCommit(ctx context.Context, fileRequest *pb.FindFilesInCommitReq
 	files, response, err := client.Contents.List(ctx, fileRequest.GetSlug(), fileRequest.GetPath(), ref, getCustomListOptsFindFilesInCommit(ctx, fileRequest))
 	if err != nil {
 		log.Errorw("FindFilesInCommit failure", "provider", gitclient.GetProvider(*fileRequest.GetProvider()), "slug", fileRequest.GetSlug(), "ref", ref, "elapsed_time_ms", utils.TimeSince(start), zap.Error(err))
-		return nil, err
+		if (response == nil) {
```
Comment (shubham149) [inline product/ci/scm/file/file.go:395]: No brackets are required i.e.
if response == nil {
}

**Comment by mohitg0795** [comment]
*[`product/ci/scm/file/file.go` line 395 at r2](https://reviewable.io/reviews/harness/harness-core/39453#-NFjlqn6RSFxXPXi0fpi-r2-395:-NFqwM2w25v5shpN_lm2:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/3f32f0a2ce3ea0d0ecdbd59162d678572311fcc1/product/ci/scm/file/file.go#L395)):*
<details><summary><i>Previously, shubham149 (Shubham Agrawal) wrote…</i></summary><blockquote>

No brackets are required i.e.
if response == nil {
}
</blockquote></details>

done

**Review by shubham149 — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39453#-:-NFqyeeo33EN-3RiHFQ9:bnfp4nl)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Comment by AnshuH** [comment]
trigger ti1 ti3 ti4

**Comment by AnshuH** [comment]
trigger ti1

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39453)* status: 12 file(s) to review. 1 of 2 approvals obtained.

**Comment by mbatchelor** [comment]
trigger ti0

**Review by adivishy1 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39453)* status: 12 file(s) to review.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39453)* status: :shipit: complete! ✓

**Pull request merged**